pytensor/link/mlx/dispatch/core.py

```diff
@@ -0,0 +1,253 @@
+"""
+pytensor/link/mlx/dispatch/basic.py
+-----------------------------------
+
+First-cut MLX translations for the most common tensor Ops.
+
+The structure intentionally follows pytensor's JAX dispatcher so that
+once these kernels stabilise they can be optimised further (e.g. fusing
+element-wise graphs, adding in-place updates, RNG thinning, etc.).
+"""
+
+from __future__ import annotations
+
+import warnings
+
+import mlx.core as mx  # MLX
+import numpy as np
+
+from pytensor.link.mlx.dispatch.basic import mlx_funcify  # MLX
+from pytensor.tensor import get_vector_length
+from pytensor.tensor.basic import (
+    Alloc,
+    AllocEmpty,
+    ExtractDiag,
+    Eye,
+    Join,
+    MakeVector,
+    ScalarFromTensor,
+    Split,
+    TensorFromScalar,
+    Tri,
+    get_scalar_constant_value,
+)
+from pytensor.tensor.exceptions import NotScalarConstantError
+
+
+# ------------------------------------------------------------------
+# Join
+# ------------------------------------------------------------------
```

Comment:
Don't introduce comments like this, the code is readable and won't get stale if we move things around